source/DistanceMap.cpp

```diff
@@ -249,7 +250,8 @@ void DistanceMap::Init(const Ship *ship)
 					// the wormhole and both endpoint systems. (If this is a
 					// multi-stop wormhole, you may know about some paths that
 					// it takes but not others.)
-					if(ship && !object.GetPlanet()->IsAccessible(ship))
+					if(!ship || (!object.GetPlanet()->IsAccessible(ship) ||
+							ship->IsRestrictedFrom(*object.GetPlanet())))
```

Comment:
Shouldn't it be `ship && (planet is inaccessible || planet is restricted)`? `!ship || ...` changes the behavior of this if statement.

Reply:
shouldnt we skip if there's no ship anyway?
```suggestion
					if(ship && (!object.GetPlanet()->IsAccessible(ship) ||
							ship->IsRestrictedFrom(*object.GetPlanet())))
```
I mean sure it doesnt matter anyway since checking for ship is more of a sanitary thing